department-of-veterans-affairs/caseflow
# Pamatya/appeals 18694

Resolves [APPEALS-18694](https://vajira.max.gov/browse/APPEALS-18694)
Also Resolves: [APPEALS-18737](https://vajira.max.gov/browse/APPEALS-18737)
Also Resolves: [APPEALS-22887](https://vajira.max.gov/browse/APPEALS-22887)


# Description

- Added Intake link to the array for switch product dropdown options.
- Added Queue link to the array for switch product dropdown options.
- Added a sort logic to sort the dropdown options to appear as it is shown in the UX design.

## Acceptance Criteria
- [x] When user selects the Caseflow Intake menu option from the drop-down menu navigation, they will be directed to the Intake page.
- [x] If the user selects ‘Caseflow Intake’ menu option:
- While currently in an intake process, they will be returned to the ‘Intake Review’ page.
- While not currently in the intake process, the page will refresh the ‘Select Form’ page.
- [x] If a user selects a menu navigation option, there will be no modals displayed if a user has entered data to the screen unless they select a ‘Cancel’ link related to their current process.
- [x] The link to Intake will only appear for VHA users that have access to Intake.
- [x] "Caseflow Intake" should be the first option in switch product dropdown
- [x] "Caseflow Queue" should be the third option in switch product dropdown as shown in the UX design
- [x] Only display Queue link to VHA users that do not have the Case Details user role.
- [x] When user selects the Caseflow Queue menu option from the drop-down menu navigation, they will be directed to the Caseflow Queue ‘Your Cases’ page.
- [x] If the user selects ‘Caseflow Queue’ menu option:
- While currently on a Case Details page, they will be returned to ‘Your cases’ page.  
- While on the main Caseflow Queue landing page, the page will refresh the ‘Your cases’ page.
- [x] If a user selects a menu navigation option, there will be no modals displayed if a user has entered data to the screen unless they select a ‘Cancel’ link related to their current process.
- [ ] Ordering of all the options should be same as shown in the UX design. 

## Testing Plan
1. GoTo /queue check if "Switch Product" dropdown already exists or not.
2. Switch users to any of the following css_id: "ACBAUERVVHAH", "PTBRADFAVBAS", "THOMAW2VACO" and go to /decision_reviews/vha or /intake you should be able to see the "Switch Product" dropdown.
3. User setup: 
```
new_user = FactoryBot.create(:user, roles: ["Reader", "Build HearSched"] )
vha_business_line = BusinessLine.find_by_url("vha")
vha_business_line.add_user(new_user)
```
Use this user to validate if the switch product is present. even though this user is a part of VHA, since this user doesn't have "Mail Intake" or "Admin Intake", "Caseflow Intake" Should not be present in the dropdown.
 
4. Add "Case Details" roles to the user. "More setup work"
```
new_user = FactoryBot.create(:user, roles: ["Case Details", "Mail Intake", "Reader", "Build HearSched"] )
vha_business_line = BusinessLine.find_by_url("vha")
vha_business_line.add_user(new_user)

```
reload and switch user to the newly created user. 
Go to  /queue -> you should be redirected to /search with Switch product dropdown but it should not contain "Caseflow Queue"
Go to /decision_reviews/vha -> you should see the dropdown 'Switch Product' but "Caseflow Queue" should not be present. 
5. 

# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
 BEFORE|AFTER
 ---|---
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/d5a2fcd4-4576-4502-ae6d-00010f4ac677)|![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/5d3c896c-daf9-4e6d-8ade-d87d25c226b1)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/325064ce-5823-4a84-997c-bfc5d8f136a4)|![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/96d972c0-13b3-41d6-bc0e-534292664964)
